mkdocs/docs/backends/cpu-dylib.md

```diff
@@ -0,0 +1,141 @@
+# Dynamic Library CPU HAL Driver
+
+IREE supports efficient model execution on CPU. IREE uses [LLVM][llvm] to
+compile dense computation in the model into highly optimized CPU native
+instruction streams, which are embedded in IREE's deployable format as dynamic
+libraries (dylibs). IREE uses its own low-overhead minimal dynamic library
+loader to load them and then schedule them with concrete workloads onto various
+CPU cores.
+
+!!! todo
+
+    Add IREE's CPU support matrix: what architectures are supported; what
+    architectures are well optimized; etc.
+
+<!-- TODO(??): when to use CPU vs GPU vs other backends -->
+
+## Get runtime and compiler
+
+### Get IREE runtime with dylib HAL driver
+
+You will need to get an IREE runtime that supports the dylib HAL driver
+so it can execute the model on CPU via dynamic libraries containing native
+CPU instructions.
+
+<!-- TODO(??): vcpkg -->
+
+
+#### Build runtime from source
+
+Please make sure you have followed the [Getting started][get-started] page
+to build IREE for Linux/Windows and the [Android cross-compilation][android-cc]
+page for Android. The dylib HAL driver is compiled in by default on all
+platforms.
+
+<!-- TODO(??): a way to verify dylib is compiled in and supported -->
+
+If you want to explicitly specify HAL drivers to support, you will need to add
+`DyLib` to the `IREE_HAL_DRIVERS_TO_BUILD` CMake list variable when configuring
+(for target).
+
+### Get compiler for CPU native instructions
+
+<!-- TODO(??): vcpkg -->
+
+#### Download as Python package
+
+IREE uses continuous integration services to compile and publish artifacts into
+[GitHub Releases][iree-releases] page. Right now these are just snapshots of
+the `main` development branch.
```

Comment:
nit: how we publish to GitHub Releases is a development detail. This could just say
> Python packages are regularly published on IREE's [GitHub Releases] page.

* "continuous integration services" could be replaced with "a random developer's machine whenever they feel like it" and users shouldn't really notice
* "just snapshots of the `main` development branch" is sort of true... a new draft release is promoted roughly once a week, but we also don't have a release branch or other process

(I see this is copied from the GPU page)